rafthttp/msg_codec.go

```diff
@@ -41,12 +42,20 @@ type messageDecoder struct {
 	r io.Reader
 }
 
+var (
+	readBytesLimit     uint64 = 512 * 1024 // 512 MB
```

Comment:
@nekto0n  We cannot do that for now. There is a 4MB limit on message size, 512KB is clearly not large enough. We set this to 512MB mainly for avoiding corrupted data to put down the server by asking it to allocate very large array.